yaml/src/Data/Yaml/Internal.hs

```diff
@@ -217,6 +219,50 @@ parseAll = do
             _ -> missed documentStart
     missed event = liftIO $ throwIO $ UnexpectedEvent event Nothing
 
+-- | Parse a yaml file (as a stream of events) to a stream of values.
+--   It only accepts documents whose top-level entity is an array.
+--
+--   In combination with 'Y.decodeFile', it can be used to consume
+--   a yaml file that consists in a very large list of values.
+parseStream :: ReaderT JSONPath (ConduitM Event Value Parse) ()
+parseStream = do
+    streamStart <- lift CL.head
+    case streamStart of
+        Nothing ->
+            -- empty string input
+            return ()
+        Just EventStreamStart ->
+            -- empty file input, comment only string/file input
+            parseDocs
+        _ -> missed streamStart
+  where
+    parseDocs = do
+        documentStart <- lift CL.head
+        case documentStart of
+            Just EventStreamEnd -> return ()
+            Just EventDocumentStart -> do
+                parseSubStream
+                requireEvent EventDocumentEnd
+                parseDocs
+            _ -> missed documentStart
+    missed event = liftIO $ throwIO $ UnexpectedEvent event Nothing
+
+parseSubStream :: ReaderT JSONPath (ConduitM Event Value Parse) ()
+parseSubStream = do
+    me <- lift CL.head
+    case me of
+      Just (EventSequenceStart _ _ a) -> parseArrayStreaming 0 a
+      _ -> liftIO $ throwIO $ UnexpectedEvent me $ Just $ EventSequenceStart NoTag AnySequence Nothing
```

Comment:
I added some values here for the "expected" event, hoping that would be more helpful that using `Nothing`.